### PR TITLE
feat(dart): improve sign in experience

### PR DESCRIPTION
**Pull request by julian-hartl**
## Summary

- Added custom app bar with back button to webview provider
- Added loading indicator to webview
- Added primary and background color property to webview
- Moved sign in uri initialization to webview to improve launch time of webview
- Made http client optional

## Testing

- Manual test

### CODE REVIEW

**Comment by julian-hartl** [comment]
@simeng-li Would you mind to give me a quick guide on how to set up logto for your flutter application?

**Comment by simeng-li** [comment]
> @simeng-li Would you mind to give me a quick guide on how to set up logto for your flutter application?

Sorry for the late response. Absolutely,
1. Set up the logto server locally. We have a [get start doc ](https://docs.logto.io/docs/tutorials/get-started/#local). Either docker image or docker-compose will do. Just follow up on the steps
2. Once the server running, you should be able to access the admin-console site under http://localhost:3001/console. Following the instruction create a native application for your flutter app. 
3. We should have a detailed app creation and integration guard.  
4. Everything we need for setting up the flutter SDK client config will be found in the admin console. Everything else left will be the same as the flutter example. 

First, let me know how it does to bring up logto locally. I'll try to come up with a detailed SDK integration guide by the end of this week.

**Comment by julian-hartl** [comment]
@simeng-li Okay, thank you very much. I got it up and running without any problems. My google sign in does not seem to work though. I get a 403 disallowed_useragent. I guess that has something to do with the browser, that the webview relies on, not being a trusted one. Did you face that issue as well and if you did, how did you solve it?

**Comment by simeng-li** [comment]
I might have some ideas. Working on it.

File: lib/logto_client.dart [hunk 1]
```diff
@@ -132,8 +146,63 @@ class LogtoClient {
...
+
+      if (postLogoutRedirectUri != null) {
+        await FlutterWebAuth.authenticate(
+          url: signOutUri.toString(),
+          callbackUrlScheme: postLogoutRedirectUri.scheme,
+        );
+      }
+    } finally {
+      if (_httpClient == null) httpClient.close();
+    }
```
Comment (simeng-li) [inline lib/logto_client.dart:206]: Even with the Nullable postLogoutRedirectUri, we intend to run the Logto signout flow. We need to clean up the sign-in session under the logto domain as well.  Is it possible to call `FlutterWebAuth.authenticate` with a empty callbackUrlScheme?

**Review by simeng-li — APPROVED** [review]
overall LGTM. Thank you so much. It works perfectly.

File: lib/logto_client.dart (hunk 1 above)
Comment (julian-hartl) [reply lib/logto_client.dart:206]: That could be why my sign in is not working. Because it is not cleaned up properly.

Comment (julian-hartl) [reply lib/logto_client.dart:206]: Couldn't you just make a HTTP call to the url.

Comment (simeng-li) [reply lib/logto_client.dart]: I'm afraid that won't work. We need to clean up the logto session cookie for those webview pages. I guess it depends on whether the `FlutterWebAuth.authenticate` stores the cookie info on ios and android or not. 
However, you are right on native apps, truly we should not have to open a new webview with the signout endpoint to clean them. There should be a better way to clean the cookies of webviews directly without navigating. 

 Let me clean this up.

**Pull request merged**